src/frontend/pages/article/table_of_contents.rs

```diff
@@ -0,0 +1,160 @@
+#[derive(PartialEq)]
```

Comment:
You should turn this into a component.

https://book.leptos.dev/view/01_basic_component.html